Makefile

```diff
@@ -267,7 +267,7 @@ GOFILES:=$(shell go list -f '{{$$p := .}}{{range $$f := .GoFiles}}{{$$p.Dir}}/{{
 .PHONY: gocyclo
 gocyclo:
 	# Run gocyclo over all .go files
-	gocyclo -over 17 ${GOFILES}
+	gocyclo -over 19 ${GOFILES}
```

Comment:
gocyclo is an indicator of over-complicated function logic. Which function is it warning on? Let's try to break down that function if possible, and only increase the limit if necessary.